executor/src/witgen/machines/block_machine.rs

```diff
@@ -140,7 +146,7 @@ fn try_to_period<T: FieldElement>(
 impl<'a, T: FieldElement> Machine<'a, T> for BlockMachine<'a, T> {
     fn process_plookup(
         &mut self,
-        fixed_data: &'a FixedData<T>,
```

Comment:
I think this should actually be removed from `Machine::process_lookup` (in a future PR). If a machine needs a reference to fixed data, it can get it at instantiation time and remember it.